go/fn/object_test.go

```diff
@@ -213,6 +213,49 @@ metadata:
 	}
 }
 
+func TestSetSlice(t *testing.T) {
+	var original = []byte(`apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: read-pods
+  namespace: default
+subjects:
+- kind: User
+  apiGroup: testing.group
+`)
+
+	var input = []byte(`apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: read-pods
+  namespace: default
+subjects:
+- kind: User
+  apiGroup: rbac.authorization.k8s.io
+- kind: Admin
+  apiGroup: rbac.authorization.k8s.io
+`)
+
+	var expected = `apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: read-pods
+  namespace: default
+subjects:
+- kind: User
+  apiGroup: rbac.authorization.k8s.io
+- kind: Admin
+  apiGroup: rbac.authorization.k8s.io
+`
+	originalObj, _ := ParseKubeObject(original)
+	inputObj, _ := ParseKubeObject(input)
+	inputSlice := inputObj.GetSlice("subjects")
+	if err := originalObj.SetSlice(inputSlice, "subjects"); err != nil {
+		t.Errorf("get slice error")
```

Comment:
typo: get -> set?

Reply:
The example is to get subObject from resource A and set it to resource B. So it is get here.

Reply:
hmmm, I don't quite follow. "get" error (if GetSlice returns errors) supposes to raise in ln 252, right?

Reply:
`GetSlice` only panics, it does not return error

Reply:
We can have the testify library handle this for us so that we don't need to worry about it ourselves.

```
require.NoError(t, originalObj.SetSlice(slicesToAdd, "subjects")
```

The import path for the `require` lib is `"github.com/stretchr/testify/require"`

Reply:
thanks, that is super helpful